image.py

```diff
@@ -18,16 +18,22 @@ class ImageManagement(object):
     def __init__(self):
 
         self.s3 = None
-        self.session = boto3.Session(aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY)
-        self.temp_dir = tempfile.TemporaryDirectory()
+        self.temp_dir = None
+        self.session = boto3.Session(
+            aws_access_key_id=AWS_ACCESS_KEY,
+            aws_secret_access_key=AWS_SECRET_KEY
+        )
 
     def download(self, image_url, image_name):
         # create temp directory to save a file in it
+        self.temp_dir = tempfile.TemporaryDirectory()
         r = requests.get(image_url, stream=True)
         if r.status_code == 200:
             im = Image.open(io.BytesIO(r.content))
-            im.save(f"{self.temp_dir}/{image_name}")
-            print(self.temp_dir)
+            with self.temp_dir as tempdir:
+                im.save(f"{tempdir}/{image_name}")
+                self.upload(f"{tempdir}/{image_name}", BUCKET_NAME)
+                print(tempdir)
```

Comment:
boss, I am uploading the directory and the image in the bucket, I need to solve this issue